azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/parser/ConfigParser.java

```diff
@@ -114,18 +110,19 @@ public IngressConfig getIngressConfig(IngressMavenConfig ingressMavenConfig) {
     }
 
     public ContainerRegistryConfig getRegistryConfig() {
+        final String defaultRegistryName = String.format("acr%s", timestamp);
+        ContainerRegistryConfig config = new ContainerRegistryConfig();
         if (mojo.getRegistry() == null) {
-            return null;
+            final ContainerRegistry registry = Azure.az(AzureContainerRegistry.class).registry(mojo.getSubscriptionId())
+                .listByResourceGroup(mojo.getResourceGroup()).stream().filter(ContainerRegistry::isAdminUserEnabled).findFirst().orElse(null);
+            config.setRegistryName(Optional.ofNullable(registry).map(ContainerRegistry::getName).orElse(defaultRegistryName));
         }
-        ContainerRegistryConfig config = new ContainerRegistryConfig();
-
-        final String defaultRegistryName = String.format("acr%s", timestamp);
-
-        config.setRegistryName(Optional.ofNullable(mojo.getRegistry().getRegistryName()).orElse(defaultRegistryName));
-        mojo.getRegistry().setRegistryName(config.getRegistryName());
-        config.setResourceGroup(Optional.ofNullable(mojo.getRegistry().getResourceGroup()).orElse(mojo.getResourceGroup()));
-        config.setSubscriptionId(Optional.ofNullable(mojo.getRegistry().getSubscriptionId()).orElse(mojo.getSubscriptionId()));
-        config.setRegion(Optional.ofNullable(mojo.getRegistry().getRegion()).orElse(mojo.getRegion()));
+        else {
+            config.setRegistryName(mojo.getRegistry().getRegistryName());
+        }
+        config.setResourceGroup(Optional.ofNullable(mojo.getRegistry()).map(ContainerRegistryConfig::getResourceGroup).orElse(mojo.getResourceGroup()));
+        config.setSubscriptionId(Optional.ofNullable(mojo.getRegistry()).map(ContainerRegistryConfig::getSubscriptionId).orElse(mojo.getSubscriptionId()));
+        config.setRegion(Optional.ofNullable(mojo.getRegistry()).map(ContainerRegistryConfig::getRegion).orElse(mojo.getRegion()));
```

Comment:
When there are multiple such nullable check, my gut feeling is that it's better to put them in the above if / else branch. The logic would be much clearer.

```java
ContainerRegistryConfig mojoRegistry = mojo.getRegistry();
if (mojoRegistry == null) {
    // set with mojoRegistry
} else {
    // set with mojo property
}
```